src/main/java/com/aws/greengrass/deployment/DefaultDeploymentTask.java

```diff
@@ -194,6 +195,18 @@ public DeploymentResult call() throws InterruptedException {
         Optional<Set<String>> groupsForDeviceOpt;
         try {
             groupsForDeviceOpt = thingGroupHelper.listThingGroupsForDevice(retryCount);
+        } catch(GreengrassV2DataException e) {
+            if (e.statusCode() == 403) {
```

Comment:
Just use AccessDeniedException?

Reply:
`GreengrassV2DataException` is the reported error and it includes status code unfortunately